.github/workflows/e2e.yaml

```
@@ -0,0 +1,75 @@
+name: e2e
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+concurrency:
+  # Cancel any running workflow for the same branch when new commits are pushed.
+  # We group both by ref_name (available when CI is triggered by a push to a branch/tag)
+  # and head_ref (available when CI is triggered by a PR).
+  group: "e2e-${{ github.ref_name }}-${{ github.head_ref }}"
+  cancel-in-progress: true
+
+jobs:
+  regular-path:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18.15.0
+          cache: yarn
+      - name: Install Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19.8
+      - run: yarn --frozen-lockfile
+      - run: make build
+      - name: Cypress run
+        uses: cypress-io/github-action@v5
+        with:
+          wait-on: http://localhost:4100/ui/
+          start: make run
+          config-file: cypress/ci.ts
+        env:
+          ELECTRON_ENABLE_LOGGING: 1
+      - uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: regular-path-cypress-screenshots
+          path: cypress/screenshots
+
+  base-path:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18.15.0
+          cache: yarn
+      - name: Install Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19.8
```

To match the other pipelines

```suggestion
          go-version: "1.19.9"
```

---

Half of the work is just building the ui + binary :( Not sure what can be done there

---

I think the `cache: true` might already shave off quite a bit